src/main/java/echoquery/sql/SchemaInferrer.java

```diff
@@ -102,20 +105,44 @@ private void adjustColumnToTable(String table, DatabaseMetaData md)
 
   /**
    *
+   * @param columns the columns of interest
+   * @param sourceTable the table of interest
+   * @param destinationTable the table being joined
+   * @return an ordered list of which table each column belongs to. the ith
+   *         table in prefixes corresponds to the ith column in columns.
+   */
+  private String[] inferPrefixes(String[] columns, String sourceTable, String destinationTable) {
+    String[] prefixes = new String[columns.length];
+    for (int i = 0; i < columns.length; i++) {
+      String column = columns[i];
+      Set<String> tables = columnToTable.get(column);
```

Comment:
I think this accounts for that. We say get every table that has this column name. Since this is after we've found a viable table to join with, that means each column name is in either the source table, the destination table, or both. the only interesting case is if its both, because otherwise we have 1 option for the prefix. in that case we choose source table with the assumption that if you say something like what is the highest id of employees with salary 10 you want employee id not job id

    what are the